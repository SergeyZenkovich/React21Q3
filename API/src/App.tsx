import React, {useEffect, useState} from "react";
import {Switch, Route} from "react-router-dom";
import styled from "styled-components";

import HomePage from "./Components/HomePage/HomePage";
import AboutPage from "./Components/AboutPage/AboutPage";
import Header from "./Components/Header/Header";
import ErrorBlock from "./Components/ErrorComponent/ErrorComponent";
import DetailsPage from "./Components/DetailsPage/DetailsPage";
import {FetchObject} from "./interfaces/ComponentsInterfaces";

const MainComponent = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const App = (): JSX.Element => {
  const [activePage, setActivePage] = useState("Home");
  const [pageElements, setPageElements] = useState([] as FetchObject[]);

  return (
    <div>
      <Header activePage={activePage} setActivePage={setActivePage} />
      <MainComponent>
        <Switch>
          <Route exact path="/about" render={() => <AboutPage setActivePage={setActivePage} />} />
          <Route
            exact
            path="/details/:id"
            render={() => <DetailsPage setActivePage={setActivePage} />}
          />
          <Route
            exact
            path="/"
            render={() => (
              <HomePage setActivePage={setActivePage} setPageElements={setPageElements} />
            )}
          />
          <Route path="/*" render={() => <ErrorBlock />} />
        </Switch>
      </MainComponent>
    </div>
  );
};
export default App;
