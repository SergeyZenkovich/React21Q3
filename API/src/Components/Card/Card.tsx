import React, {useEffect, useState} from "react";
import styled from "styled-components";
import {CardInterface, FetchObject} from "../../interfaces/ComponentsInterfaces";

import description from "../../assets/img/description.png";
import likeImg from "../../assets/img/like.png";
import name from "../../assets/img/name.png";
import place from "../../assets/img/location.png";
import tw from "../../assets/img/twitter.png";
import inst from "../../assets/img/instagram.png";
import heart from "../../assets/img/heart.png";
import photos from "../../assets/img/photos.png";
import prodImg from "../../assets/img/prod.png";

import {
  StyledCard,
  StyledCardBody,
  StyledParametersBlock,
  StyledCardMiniImg,
  StyledSeparatorBlock
} from "./CardsStyling";

const Card = ({id, color, alt_description, likes, urls, user}: FetchObject): JSX.Element => {
  const [cardDescription, setcardDescription] = useState(alt_description);
  const [cardLikes, setcardLikes] = useState(likes);
  // const [cardUrl, setcardUrl] = useState(urls.regular);
  const [cardUser, setcardUser] = useState(user);
  useEffect(() => {
    console.log("second");
    window.localStorage.setItem("cardDescription", cardDescription);
    window.localStorage.setItem("cardLikes", String(cardLikes));
    // window.localStorage.setItem("cardUrl", cardUrl);
    window.localStorage.setItem("cardUser", JSON.stringify(cardUser));
  }, [cardDescription, cardLikes, cardUser]);
  useEffect(() => {
    console.log("first");
    setcardDescription(JSON.parse(window.localStorage.getItem("cardDescription")));
    setcardLikes(Number(JSON.parse(window.localStorage.getItem("cardLikes"))));
    // setcardUrl(JSON.parse(window.localStorage.getItem("cardUrl")));
    setcardUser(JSON.parse(window.localStorage.getItem("cardUser")));
  }, []);
  return (
    <StyledCard>
      <StyledCardBody>
        <StyledParametersBlock>
          {/* <img src={urls.regular} alt="image" /> */}
          <StyledSeparatorBlock>Image info</StyledSeparatorBlock>
          {alt_description ? (
            <div>
              <StyledCardMiniImg src={description} alt="description" />
              <span>{alt_description}</span>
            </div>
          ) : null}
          <div>
            <StyledCardMiniImg src={likeImg} alt="like logo" />
            <span>{likes}</span>
          </div>
          <StyledSeparatorBlock>User info</StyledSeparatorBlock>
          {user.username ? (
            <div>
              <StyledCardMiniImg src={name} alt="priority logo" />
              <span>{user.username}</span>
            </div>
          ) : null}
          {user.location ? (
            <div>
              <StyledCardMiniImg src={place} alt="priority logo" />
              <span>{user.location}</span>
            </div>
          ) : null}
          {user.twitter_username ? (
            <div>
              <StyledCardMiniImg src={tw} alt="priority logo" />
              <span>{user.twitter_username}</span>
            </div>
          ) : null}
          {user.instagram_username ? (
            <div>
              <StyledCardMiniImg src={inst} alt="priority logo" />
              <span>{user.instagram_username}</span>
            </div>
          ) : null}
          {user.total_photos ? (
            <div>
              <StyledCardMiniImg src={photos} alt="priority logo" />
              <span>{user.total_photos}</span>
            </div>
          ) : null}
          {user.total_likes ? (
            <div>
              <StyledCardMiniImg src={heart} alt="priority logo" />
              <span>{user.total_likes}</span>
            </div>
          ) : null}
        </StyledParametersBlock>
      </StyledCardBody>
    </StyledCard>
  );
};
export default Card;
