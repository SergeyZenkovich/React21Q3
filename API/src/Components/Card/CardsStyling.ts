import styled from "styled-components";
import device from "../../utils/sizes";

import fontH from "../../assets/fonts/BebasNeue-Regular.ttf";
import fontP from "../../assets/fonts/MavenPro-Regular.ttf";

export const StyledCard = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 40%;
  @media ${device.mobileS} {
    min-width: 300px;
  }
  @media ${device.laptop} {
    min-width: 375px;
  }
  height: 50%;
  min-height: 400px;
  margin: 15px 5px;
`;
// export const StyledCardHead = styled.div`
//   @font-face {
//   font-family: "CardHeadingFont";
//     src: url(${fontH})
// }
//   display: flex;
//   justify-content: center;
//   width: 100%;
//   height: 5%;
//   background-color: ${({priority}: {priority: string}) =>
//     priority === "low" ? "#a6cd72" : priority === "medium" ? "#e2d100" : "#b82f34"};
//   & h3 {
//     color: #fff;
//     font-family: 'CardHeadingFont';
//     font-size: 1.25em;
//     margin 2% 0;
//   }
// `;
export const StyledCardBody = styled.div`
  display: flex;
  @media ${device.mobileS} {
    flex-direction: column;
  }
  @media ${device.laptop} {
    flex-direction: row;
  }
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 10px 0;
  background-color: #f5f8f9;
`;
export const StyledSeparatorBlock = styled.div`
  @font-face {
    font-family: "CardTextFont";
    src: url(${fontP});
  }
  display: flex;
  justify-content: center;
  padding: 10px 0;
  border-bottom: 2px solid #000;
  font-family: "CardTextFont";
  fonxt-size: 22px;
`;
export const StyledParametersBlock = styled.div`
  @font-face {
    font-family: "CardTextFont";
    src: url(${fontP});
  }
  display: flex;
  flex-direction: column;
  @media ${device.mobileS} {
    width: 98%;
  }
  @media ${device.laptop} {
    width: 70%;
  }
  & div {
    display: flex;
    align-items: center;
    margin: 10px 0;
  }
  & div span {
    font-family: "CardTextFont";
  }
`;
export const StyledCardImage = styled.img`
  width: 50%;
  @media ${device.mobileS} {
    min-width: 150px;
  }
  @media ${device.laptop} {
    min-width: 90px;
  }
`;
export const StyledCardMiniImg = styled.img`
  width: 30px;
  height: 30px;
  @media ${device.mobileS} {
    margin: 0 5px;
  }
  @media ${device.laptop} {
    margin: 0 10px;
  }
`;
