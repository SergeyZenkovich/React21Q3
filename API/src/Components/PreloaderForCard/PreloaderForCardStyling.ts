import React from "react";
import styled from "styled-components";

export const StyledPreloaderBlock = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 70vh;
`;
