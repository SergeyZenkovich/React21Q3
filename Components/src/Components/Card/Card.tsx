import React from "react";
import styled from "styled-components";
import {CardInterface} from "../../interfaces/ComponentsInterfaces";

import priceImg from "../../assets/img/price.png";
import dateImg from "../../assets/img/calendar.png";
import likeImg from "../../assets/img/like.png";
import ratingImg from "../../assets/img/star.png";

import {
  StyledCard,
  StyledCardHead,
  StyledCardBody,
  StyledParametersBlock,
  StyledCardImage,
  StyledCardMiniImg
} from "./CardsStyling";

const Card = ({
  cardName,
  cardPrice,
  cardYear,
  cardLikes,
  cardRating,
  cardImg
}: CardInterface): JSX.Element => (
  <StyledCard>
    <StyledCardHead>
      <h3>{cardName}</h3>
    </StyledCardHead>
    <StyledCardBody>
      <StyledCardImage src={cardImg} alt="phoneImage" />
      <StyledParametersBlock>
        <div>
          <StyledCardMiniImg src={priceImg} alt="price logo" />
          <span>Price: {cardPrice}$</span>
        </div>
        <div>
          <StyledCardMiniImg src={dateImg} alt="calendar logo" />
          <span>Release: {cardYear}</span>
        </div>
        <div>
          <StyledCardMiniImg src={likeImg} alt="like logo" />
          <span>Likes: {cardLikes}</span>
        </div>
        <div>
          <StyledCardMiniImg src={ratingImg} alt="rating logo" />
          <span>Rating: {cardRating}</span>
        </div>
      </StyledParametersBlock>
    </StyledCardBody>
  </StyledCard>
);
export default Card;
