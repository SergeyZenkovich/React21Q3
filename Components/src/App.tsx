import React from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const Cards = cardsData.map((el: CardInterface) => (
    <Card
      key={el.cardName}
      cardName={el.cardName}
      cardPrice={el.cardPrice}
      cardYear={el.cardYear}
      cardLikes={el.cardLikes}
      cardRating={el.cardRating}
      cardImg={el.cardImg}
    />
  ));
  return (
    <div>
      <Serach />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
