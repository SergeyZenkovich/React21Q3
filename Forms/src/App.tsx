import React, {useState} from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Form from "./Components/Form/Form";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const [cards, setCards] = useState([] as CardInterface[]);
  const addCard = (newCard: CardInterface): void => {
    setCards([...cards, newCard]);
  };
  const Cards = cards.map((el: CardInterface) => (
    <Card
      key={el.cardProblem}
      cardProblem={el.cardProblem}
      cardDate={el.cardDate}
      cardPriority={el.cardPriority}
      cardisRequestNeeded={el.cardisRequestNeeded}
    />
  ));
  return (
    <div>
      <Form addCard={addCard} />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
