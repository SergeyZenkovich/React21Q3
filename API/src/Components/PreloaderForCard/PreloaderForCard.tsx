import React from "react";
import styled from "styled-components";

import loaderImage from "../../assets/img/circles.gif";
import {StyledPreloaderBlock} from "./PreloaderForCardStyling";

const PreloaderForCard = (): JSX.Element => {
  return (
    <StyledPreloaderBlock>
      <img src={loaderImage} alt="gifLoader" />
    </StyledPreloaderBlock>
  );
};

export default PreloaderForCard;
