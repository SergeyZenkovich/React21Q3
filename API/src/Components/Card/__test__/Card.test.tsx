import React, {useState} from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import styled from "styled-components";
import {act} from "react-dom/test-utils";
import PreloaderForCard from "src/Components/PreloaderForCard/PreloaderForCard";
import App from "../../../App";
import Card from "../Card";
import {StyledParametersBlock, StyledSeparatorBlock, StyledCardMiniImg} from "../CardsStyling";

const testObject = {
  isLoading: false,
  element: {
    id: "testString",
    color: "#fff",
    alt_description: "test description",
    likes: 0,
    urls: {
      full: "fullTest",
      regular: "regularTest",
      small: "smallTest",
      thumb: "thumbTest"
    },
    user: {
      first_name: "testName",
      last_name: "testSurname",
      location: "testLocation",
      username: "testUserName",
      twitter_username: "testTwitter",
      instagram_username: "testIntagram",
      total_photos: 1,
      total_likes: 10
    }
  }
};
const testObject2 = {
  isLoading: true,
  element: {
    id: "testString",
    color: "#fff",
    alt_description: "test description",
    likes: 0,
    urls: {
      full: "fullTest",
      regular: "regularTest",
      small: "smallTest",
      thumb: "thumbTest"
    },
    user: {
      first_name: "testName",
      last_name: "testSurname",
      location: "testLocation",
      username: "testUserName",
      twitter_username: "testTwitter",
      instagram_username: "testIntagram",
      total_photos: 1,
      total_likes: 1
    }
  }
};
const testObjectWrong = {
  isLoading: false,
  element: {
    id: null,
    color: null,
    alt_description: null,
    likes: 0,
    urls: {
      full: "fullTest",
      regular: "regularTest",
      small: "smallTest",
      thumb: "thumbTest"
    },
    user: {
      first_name: "testName",
      last_name: "testSurname",
      location: "testLocation",
      username: "testUserName",
      twitter_username: "testTwitter",
      instagram_username: "testIntagram",
      total_photos: 1,
      total_likes: NaN
    }
  }
};
const testObjectWrong2 = {
  isLoading: false,
  element: {
    id: "testId",
    color: null,
    alt_description: null,
    likes: 0,
    urls: {
      full: "fullTest",
      regular: "regularTest",
      small: "smallTest",
      thumb: "thumbTest"
    },
    user: {
      first_name: "testName",
      last_name: "testSurname",
      location: "",
      username: "",
      twitter_username: "",
      instagram_username: "",
      total_photos: NaN,
      total_likes: NaN
    }
  }
};

it("renders img inside Card block without crashing", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  expect(container.querySelector("img").src).toBe(
    `http://localhost/${testObject.element.urls.regular}`
  );
});

it("renders inner styledComponentns div", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlockClassName = StyledParametersBlock.styledComponentId;
  const innerBlock = container.querySelector(`.${innerBlockClassName}`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div on id null", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong}></Card>, container);
  const innerBlockClassName = StyledParametersBlock.styledComponentId;
  const innerBlock = container.querySelector(`.${innerBlockClassName}`);
  expect(container.contains(innerBlock)).toBeFalsy();
});

it("renders inner styledComponentns div with description", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.descriptionBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with description on wrong description", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.descriptionBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});
it("renders inner styledComponentns div with username", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.userNameBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with username on wrong username", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.userNameBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});

it("renders inner styledComponentns div with user Location", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.userLocationBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with user Location on wrong user Location", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.userLocationBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});

it("renders inner styledComponentns div with user twitter", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.twitterBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with user twitter on wrong user twitter", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.twitterBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});
it("renders inner styledComponentns div with user instagram", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.intagramBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with user instagram on wrong user instagram", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.intagramBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});

it("renders inner styledComponentns div with user total photos", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.totalPhotosBlock`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

it("not renders inner styledComponentns div with user total photos on wrong user total photos", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlock = container.querySelector(`.totalPhotosBlock`);
  expect(container.contains(innerBlock)).toBeFalsy();
});

it("renders inner styledComponentns div with usersTotalLikes", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject}></Card>, container);
  const innerBlock = container.querySelector(`.totalBlock`);
  expect(container.querySelector(`.totalBlock span`).textContent).toBe("10");
});

it("not renders inner styledComponentns div with usersTotalLikes on wrong usersTotalLikes", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObjectWrong2}></Card>, container);
  const innerBlockClassName = StyledParametersBlock.styledComponentId;
  const innerBlock = container.querySelector(`.${innerBlockClassName}`);
  const innerBlock2 = innerBlock.querySelector(`.totalBlock`);
  expect(container.contains(innerBlock2)).toBeFalsy();
});

it("render preloader if loading", () => {
  const container = document.createElement("div");
  ReactDOM.render(<Card {...testObject2}></Card>, container);
  expect(container.querySelector("img").alt).toBe(`gifLoader`);
});

describe("Card Component", () => {
  const setFunction = jest.fn();
  let container;
  beforeEach(() => {
    container = renderer.create(<Card {...testObject}></Card>);
  });
  it("matches shapshot", () => {
    const tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
