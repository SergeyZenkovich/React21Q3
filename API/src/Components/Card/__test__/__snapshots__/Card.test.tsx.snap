// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches shapshot 1`] = `
<div
  className="sc-bdnxRM hLCZwY"
>
  <div
    className="sc-gtsrHT ccuSmW"
  >
    <div
      className="sc-hKFxyN deOwSh"
    >
      <img
        alt="image"
        src="regularTest"
      />
      <div
        className="sc-dlnjwi cuMsqH"
      >
        Image info
      </div>
      <div
        className="descriptionBlock"
      >
        <img
          alt="description"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          test description
        </span>
      </div>
      <div>
        <img
          alt="like logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          0
        </span>
      </div>
      <div
        className="sc-dlnjwi cuMsqH"
      >
        User info
      </div>
      <div
        className="userNameBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          testUserName
        </span>
      </div>
      <div
        className="userLocationBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          testLocation
        </span>
      </div>
      <div
        className="twitterBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          testTwitter
        </span>
      </div>
      <div
        className="intagramBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          testIntagram
        </span>
      </div>
      <div
        className="totalPhotosBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          1
        </span>
      </div>
      <div
        className="totalBlock"
      >
        <img
          alt="priority logo"
          className="sc-jSFjdj gJUICa"
          src="test-file-stub"
        />
        <span>
          10
        </span>
      </div>
    </div>
  </div>
</div>
`;
